File: backend/src/main/java/EazyTech/EazyHire/services/FormFieldService.java
package EazyTech.EazyHire.services;

import EazyTech.EazyHire.models.dtos.FormFieldRequestDTO;
import EazyTech.EazyHire.models.dtos.FormFieldResponseDTO;

import java.util.List;

public interface FormFieldService {

    List<FormFieldResponseDTO> getFormFields(Long jobId, Long companyId);

    FormFieldResponseDTO createFormField(Long jobId, Long companyId, Long actorId, FormFieldRequestDTO request);

    FormFieldResponseDTO updateFormField(
            Long jobId,
            Long fieldId,
            Long companyId,
            Long actorId,
            FormFieldRequestDTO request
    );

    void deleteFormField(Long jobId, Long fieldId, Long companyId, Long actorId);

    List<FormFieldResponseDTO> reorderFormFields(Long jobId, Long companyId, Long actorId, List<Long> orderedIds);
}

File: frontend/src/services/form-field.service.ts
import { api } from './api';
import type { BaseResponse } from '@/types/api.types';
import type { FormField, FormFieldRequest } from '@/types/form-field.types';

export const formFieldService = {
  getFormFields: async (jobId: number | string) => (
    await api.get<BaseResponse<FormField[]>>(`/jobs/${jobId}/form-fields`)
  ).data.data,

  createFormField: async (jobId: number | string, data: FormFieldRequest) => (
    await api.post<BaseResponse<FormField>>(`/jobs/${jobId}/form-fields`, data)
  ).data.data,

  updateFormField: async (jobId: number | string, fieldId: number, data: FormFieldRequest) => (
    await api.put<BaseResponse<FormField>>(`/jobs/${jobId}/form-fields/${fieldId}`, data)
  ).data.data,

  deleteFormField: async (jobId: number | string, fieldId: number) => (
    await api.delete<BaseResponse>(`/jobs/${jobId}/form-fields/${fieldId}`)
  ).data,

  reorderFormFields: async (jobId: number | string, orderedIds: number[]) => (
    await api.put<BaseResponse<FormField[]>>(`/jobs/${jobId}/form-fields/reorder`, { orderedIds })
  ).data.data,
};
